### PR TITLE
v0.13.0 Release: Ruby Version Upgrade Part #1

### PR DESCRIPTION
**Pull request by abhinav-nain**
# Description

Changing Ruby version to 3.1 (Current least supported version of Ruby)
  * Includes upgrading various gems specified in tha gemspec

# Checklist:

- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added entries to `CHANGELOG.md` and marked it Added/Changed/Removed
- [x] I have not incremented `version.rb`

Please note- This is just ruby version upgrade, part 2 will contain the VRT Update as well.

### CODE REVIEW

**Comment by rohit-bugcrowd** [comment]
Changes Look good.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:2.5-alpine
+FROM ruby:3.1
```
Comment (sumanbalu-bugcrowd) [inline Dockerfile:1]: @abhinav-nain , Is there a reason for not using 3.1-alpine ?

**Review by sumanbalu-bugcrowd — COMMENTED** [review]
Rest looks good to me!

File: Dockerfile (hunk 1 above)
Comment (abhinav-nain) [reply Dockerfile:1]: Yes, it seems like the 3.1-alpine image has some dev tools issue and the gem installation fails

**Review by sumanbalu-bugcrowd — APPROVED** [review]

**Comment by abhinav-nain** [comment]
Blocking this as gh actions need to be reconfigured

**Pull request merged**